src/components/ui.tsx

```diff
@@ -56,57 +111,89 @@ export function Box({
   radius,
   center = false,
   order,
-  cx,
+  cx: _cx = [],
   ...props
-}) {
+}: WithChildren<BoxProps>) {
   return (
     <Base
       cx={[
         styles.widths[width],
-        styles.backgrounds[background],
-        styles.padding[padding],
-        styles.paddingY[paddingY],
-        styles.radii[radius],
+        background && styles.backgrounds[background],
+        padding && styles.padding[padding],
+        paddingY && styles.paddingY[paddingY],
+        radius && styles.radii[radius],
         center && styles.box.center,
         order && styles.order[order],
-        cx,
+        ..._cx,
       ]}
       {...props}
     />
   )
 }
 
-export function FlexList({ ...props }) {
-  return <Flex as="ul" cx={styles.list} {...props} />
+interface FlexListProps extends FlexProps {}
+
+export function FlexList(props: WithChildren<FlexListProps>) {
+  return <Flex as="ul" cx={[styles.list]} {...props} />
 }
 
 export function List(props) {
   return <Base as="ul" cx={[styles.list]} {...props} />
 }
 
-export function Space({ size = "auto", ...props }) {
+interface SpaceProps extends BaseProps {
+  size?: SpaceTokens | "auto"
+}
+
+export function Space({ size = "auto", ...props }: SpaceProps) {
   return <Base cx={[styles.margin[size]]} {...props} />
 }
 
-export function Nudge({ left, right, top, bottom, ...props }) {
+interface NudgeProps {
+  left?: number
+  right?: number
+  top?: number
+  bottom?: number
+}
+
+export function Nudge({
+  left,
+  right,
+  top,
+  bottom,
+  ...props
+}: WithChildren<NudgeProps>) {
   return (
     <Base
       cx={[
-        styles.margin.left[-left],
-        styles.margin.right[-right],
-        styles.margin.top[-top],
-        styles.margin.bottom[-bottom],
+        left && styles.marginLeft[-left],
```

Comment:
I thought these props could be `0`, but I might be wrong

Reply:
They could, but there would be no point in passing `0` as the value since you wouldn't be "nudging" at all.

Reply:
In which case, this would pass a `0` to the `className`, right?

Reply:
Ah, no I'm wrong, this goes through `.filter(Boolean)`